examples/dataflow-custom-templates/java/src/main/java/com/google/example/WordCount.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.google.example;
+
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.io.TextIO;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.Filter;
+import org.apache.beam.sdk.transforms.FlatMapElements;
+import org.apache.beam.sdk.transforms.MapElements;
+
+/**
+ * Count words read from a file blob.
+ *
+ * <p>See {@link WordCountOptions} for a list of expected arguments.
+ */
+public class WordCount {
+
+  public static void main(String[] args) {
+    WordCountOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(WordCountOptions.class);
+
+    Pipeline p = Pipeline.create(options);
+
+    p.apply("ReadFromSource", TextIO.read().from(options.getSource()))
+        .apply("SplitLinesIntoTokens", FlatMapElements.via(split()))
+        .apply("FilterNonEmptyTokens", Filter.by(filterNonEmpty()))
+        .apply("ConvertToLowerCase", MapElements.via(toLowerCase()))
+        .apply("CountPerToken", Count.perElement())
+        .apply("FormatCountsToDisplay", MapElements.via(displayCounts()))
+        .apply("OutputResult", TextIO.write().to(options.getOutput()));
+
+    p.run();
+  }
+
+  static DisplayCounts displayCounts() {
```

Comment:
Is there any specific reason for these extra static methods rather than creating instances directly inside the transforms

Reply:
@prathapreddy123 Thank you so much for reviewing. When I was first learning beam, it was helpful to see PTransforms written as separate classes rather than the lambda approach we typically see with examples.  The example code is actually a minor feature of this repository and happy to change if you feel this comment you raised is blocking to merging the PR.  Could you let us know by either resolving the conversation or replying that it is important?

(cc: @itodotimothy6 )